src/interface/window.c

```diff
@@ -2477,3 +2477,32 @@ void window_update_textbox()
 		window_event_textinput_call(w, gCurrentTextBox.widget_index, gTextBoxInput);
 	}
 }
+
+bool window_is_visible(rct_window* w)
+{
+	if (w->viewport == NULL)
+	{
+		// default to previous behavior
+		return true;
+	}
+
+	// consider main window always visible
+	if (w->classification == WC_MAIN_WINDOW)
+		return true;
+
+	for (rct_window *w_other = g_window_list; w < RCT2_NEW_WINDOW; w_other++)
+	{
+		// if covered by a higher window, no rendering needed
+		if (w_other > w
+			&& w_other->x <= w->x
+			&& w_other->y <= w->y
+			&& (w_other->x + w_other->width) >= (w->x + w->width)
+			&& (w_other->y + w_other->height) >= (w->y + w->height))
+		{
+			return false;
+		}
+	}
+
+	// default to previous behavior
+	return true;
+}
```

Comment:
VS won't automatically add a new line. So if you paste or type at the end of the document, it will be after the new line character.